### ISSUE TITLE
Make List Submodule Internal

### ISSUE THREAD

**Issue by zckeyser** [konkked/Underscore.cs#17]
There's no need to have List exposed externally, since it's exposing implementation details that aren't important to the consumer. Instead, consumers should just use Collection equivalents and if there's a relevant optimization the library will switch to it internally.

**Comment by konkked** [comment]
@zckeyser just ran into the reason why I made the list sub module, basically there are somethings that would be nonsensical to use on an enumerable, like shuffle, and certain methods that wouldn't even make sense at all (e.g. Shuffle override with option to do in place ) so might be good to rephrase this to  limit the List module to only include methods that only make sense for the list, but move the more efficient list algos into the collection object

**Comment by zckeyser** [comment]
@konkked How would you want to go about moving those? Add a new sub-module inside collection for the alternate implementations?
